parlai/mturk/core/react_server/dev/app.jsx

```diff
@@ -235,6 +238,170 @@ class MainApp extends React.Component {
   }
 }
 
-var main_app = <MainApp />;
+
+// TODO consolidate shared functionality from SocketManager in a way that
+// prevents this class from setting a whole lot of dummy methods
+class StaticApp extends React.Component {
+  constructor(props) {
+    super(props);
+    let initialization_status = 'initializing';
+    if (!doesSupportWebsockets()) {
+      initialization_status = 'websockets_failure';
+    }
+
+    // TODO move constants to props rather than state
+    this.state = {
+      task_description: null,
+      mturk_submit_url: null,
+      frame_height: FRAME_HEIGHT,
+      socket_status: null,
+      hit_id: HIT_ID, // gotten from template
+      assignment_id: ASSIGNMENT_ID, // gotten from template
+      worker_id: WORKER_ID, // gotten from template
+      conversation_id: null,
+      initialization_status: initialization_status,
+      world_state: null, // TODO cover onboarding and waiting separately
+      is_cover_page: IS_COVER_PAGE, // gotten from template
+      done_text: null,
+      chat_state: 'idle', // idle, text_input, inactive, done
+      task_done: false,
+      messages: [],
+      agent_id: 'NewWorker',
+      task_data: {},
+      response_data: {},
+      volume: 1, // min volume is 0, max is 1, TODO pull from local-storage?
+    };
+  }
+
+  handleIncomingHITData(data) {
+    let task_description = data['task_description'];
+    if (isMobile() && BLOCK_MOBILE) {
+      task_description =
+        '<h1>Sorry, this task cannot be completed on mobile devices. ' +
+        'Please use a computer.</h1><br>Task Description follows:<br>' +
+        data['task_description'];
+    }
+
+    this.setState({
+      task_description: task_description,
+      frame_height: data['frame_height'] || 650,
+      mturk_submit_url: data['mturk_submit_url'],
+    });
+  }
+
+  componentDidMount() {
+    getHitConfig(data => this.handleIncomingHITData(data));
+  }
+
+  onMessageSend(text, data, callback, is_system) {
+    if (text === '') {
+      return;
+    }
+    this.socket_handler.handleQueueMessage(text, data, callback, is_system);
+  }
+
+  onValidData(valid, response_data) {
+    this.setState({task_done: valid, response_data: response_data});
+  }
+
+  render() {
+    let socket_handler = null;
+    if (!this.state.is_cover_page) {
+      socket_handler = (
```

Comment:
this seems like it might be good to refactor into a helper method, this is a really long if

Reply:
This whole thing is going to be replaced in v1,  and really it's a single assign statement, but otherwise agreed.